asynciterator.ts

```diff
@@ -2035,25 +2192,17 @@ export function range(start: number, end: number, step?: number) {
   return new IntegerIterator({ start, end, step });
 }
 
-// Determines whether the given object is a function
-function isFunction(object: any): object is Function {
-  return typeof object === 'function';
-}
-
-// Determines whether the given object is an EventEmitter
-function isEventEmitter(object: any): object is EventEmitter {
-  return object && typeof object.on === 'function';
-}
-
-// Determines whether the given object is a promise
-function isPromise<T>(object: any): object is Promise<T> {
-  return object && typeof object.then === 'function';
+export interface WrapOptions {
+  prioritizeIterable?: boolean;
+  letIteratorThrough?: boolean;
```

Comment:
@jacoscaz Could you explain what both options are doing?

I would try to make them obsolete (my preference) or rename them (if there is no way around having them).

Reply:
Hi @RubenVerborgh !

> @jacoscaz Could you explain what both options are doing?

- `prioritizeIterable` gives users more control over what interface might be used to iterate upon a source in case the source implements more than one of the supported interfaces. An array, for example, can be wrapped both via `fromArray` and via `fromIterable`.

- `letIteratorThrough` gives users the option of having `AsyncIterator` instances passed through as they are. I've done this a while ago so I'm not 100% sure but I think I added this mostly for backward compatibility.
 
> I would try to make them obsolete (my preference) or rename them (if there is no way around having them).

Looking at this again after a while, I would also make these obsolete and just have the library do whatever's best for the given source (including passing the latter through as-is). @jeswr might want to pitch in as he initially suggested `prioritizeIterable`.

Reply:
Thanks, removed both in e83a5104b3cb0078992a146beb75ed34e96f0813.